src/dbhelper.py

```diff
@@ -0,0 +1,107 @@
+from PySide import QtSql
+from PySide.QtSql import QSqlQuery
+from passwords import DB_SERVER, DB_PORT, DB_LOGIN, DB_PASSWORD, DB_TABLE
+import logging
+
+# Object to wrap a database connection and provide some vaguely handy utilities.
+_logger = logging.getLogger(__name__)
+
+# Open the database connection. when the module is first touched...
+_db = QtSql.QSqlDatabase("QMYSQL")
+_db.setHostName(DB_SERVER)
+_db.setPort(DB_PORT)
+
+_db.setDatabaseName(DB_TABLE)
+_db.setUserName(DB_LOGIN)
+_db.setPassword(DB_PASSWORD)
+
+_db.setConnectOptions("MYSQL_OPT_RECONNECT=1")
+
+if not _db.open():
+    _logger.error(_db.lastError().text())
+    raise RuntimeError
+
+# A cache of prepared queries, keyed by SQL string. Allows reuse of prepared queries.
+_prepared_queries = {}
```

Comment:
I'm not sure it's sensible to cache the `QSqlQuery` objects themselves, as these are `PySide`-owned C++ objects.

Generally, I do think Qt or even the underlying database handles caching of prepared queries as well -- but I would need to investigate further to be sure. Did you check that out?